### PR TITLE
Feature/uppsf 2943 remove methode

### PR DESCRIPTION
**Pull request by StephanZahariev**
# Description

## What

Remove Methode dependency

## Why

[UPPSF-2943](https://financialtimes.atlassian.net/browse/UPPSF-2943)

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [X] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: content/draft_content_mapper_resolver_test.go [hunk 1]
```diff
@@ -8,53 +8,26 @@ import (
...
+	resolver := NewDraftContentMapperResolver(map[string]DraftContentMapper{})
 
 	mapper, err := resolver.MapperForContentType("application/vnd.ft-upp-article+json; version=1.0; charset=utf-8")
 
 	assert.Error(t, err)
 	assert.Nil(t, mapper)
 }
 
-func happyResolverConfig(mam DraftContentMapper, ucv DraftContentMapper) (contentTypeToMapper map[string]DraftContentMapper) {
+func happyResolverConfig(ucv DraftContentMapper) (contentTypeToMapper map[string]DraftContentMapper) {
```
Comment (atanasdinov) [inline content/draft_content_mapper_resolver_test.go:29]: This function is now identical to the `cctOnlyResolverConfig` one below and can be safely removed.

File: content/draft_content_rw_test.go [hunk 2]
```diff
@@ -203,11 +203,14 @@ func mockReadFromGenericRW(t *testing.T, status int, contentUUID string, systemI
...
 		w.Header().Add(originSystemIdHeader, systemID)
-		w.Header().Set("Content-Type", contentType)
+		w.Header().Set("Content-Type", contentTypeArticle)
 		w.Header().Set("Write-Request-Id", writeRef)
 		w.Header().Set("Last-Modified-RFC3339", lastModified)
 		w.WriteHeader(status)
-		w.Write(body)
+		_, err := w.Write(body)
+		if err != nil {
+			panic(err)
```
Comment (atanasdinov) [inline content/draft_content_rw_test.go:212]: A `panic` here would terminate the execution of all tests. Is it intended?

File: content/handler_timeout_test.go [hunk 3]
```diff
@@ -197,9 +166,12 @@ func newDraftContentRWTestServer(inducedDelay time.Duration, responseStatus int,
 		case http.MethodGet:
 			logrus.Info("Processing GET test request")
 			w.Header().Set("Content-Type", contentType)
-			w.Header().Set("X-Origin-System-Id", originId)
+			w.Header().Set("X-Origin-System-Id", originID)
 			w.WriteHeader(responseStatus)
-			w.Write([]byte(fromMaMContent))
+			_, err := w.Write([]byte(fromUppContent))
+			if err != nil {
+				panic(err)
```
Comment (atanasdinov) [inline content/handler_timeout_test.go:173]: A `panic` here would terminate the execution of all tests. Is it intended?

File: content/handler_timeout_test.go [hunk 4]
```diff
@@ -208,33 +180,19 @@ func newDraftContentRWTestServer(inducedDelay time.Duration, responseStatus int,
...
 			m.EndpointCalled()
 			time.Sleep(inducedDelay)
 		}
-		w.Header().Set("Content-Type", contentType)
+		w.Header().Set("Content-Type", contentTypeArticle)
 		w.WriteHeader(responseStatus)
-		w.Write([]byte(fromUppContent))
+		_, err := w.Write([]byte(fromUppContent))
+		if err != nil {
+			panic(err)
```
Comment (atanasdinov) [inline content/handler_timeout_test.go:194]: A `panic` here would terminate the execution of all tests. Is it intended?

File: platform/service_test.go [hunk 5]
```diff
@@ -47,7 +47,10 @@ func newGTGServerMock(t *testing.T, httpStatus int, body string) *httptest.Serve
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		assert.Equal(t, status.GTGPath, r.URL.Path)
 		w.WriteHeader(httpStatus)
-		w.Write([]byte(body))
+		_, err := w.Write([]byte(body))
+		if err != nil {
+			panic(err)
```
Comment (atanasdinov) [inline platform/service_test.go:52]: A `panic` here would terminate the execution of all tests. Is it intended?

**Review by atanasdinov — COMMENTED** [review]
Looks good to me but I'm concerned about the introduced `panic` calls. Could we double check if we need them?

**Review by atanasdinov — APPROVED** [review]

File: config.yml [hunk 6]
```diff
@@ -11,9 +11,6 @@ content-types:
   "application/vnd.ft-upp-content-placeholder+json":
     mapper: "spark"
```
Comment (MiroslavGatsanoga) [inline config.yml:12]: **thought:** ‏‏The usage of the word `mapper` here and everywhere else in the service might be confusing at first sight for those who have knowledge of Methode and associate it with it. Not to what we should rename it though, `validator` probably?

**Review by MiroslavGatsanoga — APPROVED** [review]

File: config.yml (hunk 6 above)
Comment (StephanZahariev) [reply config.yml:12]: Indeed, a lot of room for refactoring. Maybe we can open a new task since the changes will go beyond the scope of removing Methode.

**Pull request merged**